include/core/JamomaSample.h

```diff
@@ -244,7 +254,92 @@ namespace Jamoma {
 			for (int i=0; i<frameCount; ++i)
 				values[i] = vector[i];
 		}
-		
+        
+        
+        /** Duplicate values at the beginning and end to provide padding for interpolation.
+            @param  paddingAmount   Number of frames to add to both beginning & end of SampleBundle. Total number of frames added to SampleBundle will be twice this amount.
+         */
+        void applySamplePadding(int paddingAmount)
+        {
+            
+            // ensure that we are adding at least one sample, AND not more than half SampleBundle frames
+            assert(paddingAmount > 0 && paddingAmount < mFrameCount/2);
+            
+            // make a copy
+            SampleBundleData newPaddedChannels = mChannels;
+            
```

Comment:
I understand the reason for doing the copy, but it makes me a little nervous because a SampleBundle could potentially be really big, right?